google/cloud/spark_connect/client/core.py

```diff
@@ -51,7 +57,7 @@ def toChannel(self) -> grpc.Channel:
         return self._proxied_channel()
```

Comment:
I think we should add a check here that the session is still active before trying to return a channel to it.